pkg/operator/k8sutil/node.go

```diff
@@ -230,19 +236,41 @@ func GetKubernetesNodesMatchingRookNodes(rookNodes []rookalpha.Node, clientset k
 
 // RookNodesMatchingKubernetesNodes returns only the given Rook nodes which have a corresponding
 // match in the list of Kubernetes nodes.
-func RookNodesMatchingKubernetesNodes(rookNodes []rookalpha.Node, kubernetesNodes []v1.Node) []rookalpha.Node {
+func RookNodesMatchingKubernetesNodes(rookStorage rookalpha.StorageScopeSpec, kubernetesNodes []v1.Node) []rookalpha.Node {
 	nodes := []rookalpha.Node{}
 	for _, kn := range kubernetesNodes {
-		for _, rn := range rookNodes {
+		for _, rn := range rookStorage.Nodes {
 			if rookNodeMatchesKubernetesNode(rn, kn) {
 				rn.Name = normalizeHostname(kn)
+
+				// modify the node Location if topology awareness enabled
+				if rookStorage.TopologyAware == true {
+					rn.Location = nodeTopologyLocation(kn, rn.Location)
```

Comment:
Will the location loaded from the topology override the location if it is set in the cluster.yaml? We'll need to understand/document the precedence.

Reply:
If I remember right, if more than one of the same topology type is specified in the location, the latest one take precedence. @liewegas, am I making things up?

Reply:
iiuc cluster.yaml is the global ceph config, loaded in the mon config db?  i think ``crush_location`` shouldn't be touched there.

but in general, the environment variable or command line will take a higher precedence than a ceph.conf or the mon config db.

Reply:
The `location` in the cluster.yaml is essentially passed down to the ceph.conf as the `crush location` setting when an osd is created in the "osd prepare" job.

Reply:
As we finish the conversion of the daemon configs, this will likely change to a command-line option. @jarrpa As far as the precedence, the new topology aware settings will override the cluster.yaml `location`, right?

Reply:
@travisn Correct.

Reply:
@liewegas I believe cluster.yaml is more than just that. You can specify either a cluster-wide location for all nodes or a per-node location there. This just takes anything specified in the cluster config and appends any topology labels found. I don't think location is saved to either a ceph.conf or the mon config db, but I could be wrong.